src/Core/Domain/MailTemplate/CommandHandler/GenerateThemeMailTemplatesCommandHandler.php

```diff
@@ -108,25 +108,21 @@ public function handle(GenerateThemeMailTemplatesCommand $command)
     }
 
     /**
-     * When installing a new Language, if it's a new one the Translator component can't manage it because its cache is
-     * already filled with the default one as fallback. We force the component to update its cache by adding a fake
-     * resource for this locale (this is the only way clean its local cache)
+     * When installing a new language, we must add the lang pack to the translator
      *
-     * @param string $locale
+     * @param $locale
      */
-    private function cleanTranslatorLocaleCache($locale)
+    private function addLanguageToTranslator($locale)
     {
-        if (!method_exists($this->translator, 'addLoader')
-            || !method_exists($this->translator, 'addResource')
-        ) {
-            return;
+        $finder = Finder::create()
```

Comment:
You can avoid re-populating the translator if it's already done
```suggestion
		if (0 < count($this->translator->getCatalogue($locale)->all()) {
		    // avoid re-populating the translator
		    return;
		}

        $finder = Finder::create()
```

Reply:
What if we're updating the lang pack? Don't we need to update the translator with the latest translations?

Reply:
I see what you mean. But in case of the current language being updated, the resources have already been added to the translator. What happens to the translator if you add the same resource twice?

I have the feeling that we need to factorize this translator refresh functionality out of this class and have it shared with the one in Context.